### DISCUSSION TITLE
Block State Transition after making field required

### DISCUSSION THREAD

**Discussion by sykokinetyk** [tfsaggregator/aggregator-cli#230] [Q&A]
Hello all,

I'm trying to block transitions between states (or something similar to that effect) if a required field is not populated.  Are there any modules of "self" that check for state transition or destination state?  Here's what I'm working with so far, any assistance or advice is appreciated.

      if ((string)self["System.State"] == "QA") {
        if (self["Microsoft.VSTS.Scheduling.CompletedWork"] !> 0) {
          
        }

**Comment by giuliov** [comment]
Sorry for delayed reply but never received any notification.

Shortly, you cannot block a change, any change not just state, because Azure DevOps send the event to Aggregator _after_ committing the change to the database.
Your best option is to change the work item type workflow as explained [here](https://docs.microsoft.com/en-us/azure/devops/organizations/settings/work/custom-rules?view=azure-devops).
